core/capabilities/webapi/trigger.go

```diff
@@ -1,18 +1,293 @@
+					if err != nil {
+						h.lggr.Errorw("error sending response", "err", err)
+					}
+					return
+				}
+				if !trigger.rateLimiter.Allow(body.Sender) {
+					h.lggr.Errorw("request rate-limited", "sender", sender.String(), "messageID", body.MessageId)
+					err = h.sendResponse(ctx, gatewayID, body, webapicapabilities.TriggerResponsePayload{Status: "ERROR", ErrorMessage: "request rate-limited"})
+					if err != nil {
+						h.lggr.Errorw("error sending response", "err", err)
+					}
+					return
+				}
+
+				TriggerEventID := body.Sender + payload.TriggerEventID + strconv.FormatInt(time.Now().Unix(), 10)
+				tr := capabilities.TriggerResponse{
+					Event: capabilities.TriggerEvent{
+						TriggerType: TriggerType,
+						ID:          TriggerEventID,
+						Outputs:     wrappedPayload,
+					},
+				}
+				select {
+				case <-ctx.Done():
+					return
+				case trigger.ch <- tr:
+					response = webapicapabilities.TriggerResponsePayload{Status: "ACCEPTED"}
+					// Sending n topics that match a workflow with n allowedTopics, can only be triggered once.
+					break
+				}
+			}
+		}
+	}
+	if matchedWorkflows == 0 {
+		h.lggr.Errorw("No Matching Workflow Topics")
+		response = webapicapabilities.TriggerResponsePayload{Status: "ERROR", ErrorMessage: "No Matching Workflow Topics"}
+	}
+	err = h.sendResponse(ctx, gatewayID, body, response)
+	if err != nil {
+		h.lggr.Errorw("Error sending response", "body", body, "response", response, "err", err)
+	}
+}
+
+func (h *triggerConnectorHandler) HandleGatewayMessage(ctx context.Context, gatewayID string, msg *api.Message) {
+	// TODO: Validate Signature
+	body := &msg.Body
+	sender := ethCommon.HexToAddress(body.Sender)
+	var payload webapicapabilities.TriggerRequestPayload
+	err := json.Unmarshal(body.Payload, &payload)
+	if err != nil {
+		h.lggr.Errorw("error decoding payload", "err", err)
+		err = h.sendResponse(ctx, gatewayID, body, webapicapabilities.TriggerResponsePayload{Status: "ERROR", ErrorMessage: fmt.Errorf("error %s decoding payload", err.Error()).Error()})
+		if err != nil {
+			h.lggr.Errorw("error sending response", "err", err)
+		}
+		return
+	}
+
+	switch body.Method {
+	case webapicapabilities.MethodWebAPITrigger:
+		h.processTrigger(ctx, gatewayID, body, sender, payload)
+		return
+
+	default:
+		h.lggr.Errorw("unsupported method", "id", gatewayID, "method", body.Method)
+		err = h.sendResponse(ctx, gatewayID, body, webapicapabilities.TriggerResponsePayload{Status: "ERROR", ErrorMessage: fmt.Errorf("unsupported method %s", body.Method).Error()})
+		if err != nil {
+			h.lggr.Errorw("error sending response", "err", err)
+		}
+	}
+}
+
+func (h *triggerConnectorHandler) RegisterTrigger(ctx context.Context, req capabilities.TriggerRegistrationRequest) (<-chan capabilities.TriggerResponse, error) {
+	cfg := req.Config
+	if cfg == nil {
+		h.lggr.Errorw("config is required to register a web api trigger")
+		return nil, errors.New("config is required to register a web api trigger")
```

Comment:
No need to log + return the error here; it'll get logged upstream if the caller wants to (in prod, it will get logged by the workflow engine)

Reply:
Same with the other errors in the function